home/views/admin.py

```diff
@@ -167,6 +167,11 @@ def post(self, request):
                 ProfessorCourse.objects.filter(professor__id=subject_id).update(professor=merge_target)
                 Review.unfiltered.filter(professor__id=subject_id).update(professor=merge_target)
                 Grade.unfiltered.filter(professor__id=subject_id).update(professor=merge_target)
+                
+                aliases = ProfessorAlias.objects.filter(name=merge_subject.name, professor=merge_target)
+                if not aliases.exists():
+                    ProfessorAlias(name=merge_subject.name, professor=merge_target).save()
+
```

Comment:
`save` upserts, so checking for existence beforehand isn't necessary. not going to nitpick though.

Reply:
there aren't unique constraints on any ProfessorAlias fields so the check is avoid duplicates

Reply:
sure, but I'm fairly certain `ProfessorAlias(name=merge_subject.name, professor=merge_target).save()` on its own without the `.exists()` would retrieve the existing alias if it exists (in which case `save` is a no-op), or would create the alias otherwise, which is what we want.

Reply:
It appears that `save()` allows duplicates by default so the check is necessary. <img width="754" alt="Screen Shot 2022-11-12 at 6 56 20 PM" src="https://user-images.githubusercontent.com/37155082/201499235-a2719123-a090-426a-8bfe-9dc6c036da0b.png">

Reply:
huh fair enough, I was mistaken. I did read [how django knows to UPDATE vs INSERT](https://docs.djangoproject.com/en/4.1/ref/models/instances/#how-django-knows-to-update-vs-insert), but forgot that when you create a model manually, it doesn't have a pk set (which causes django to follow the second bullet point and perform an INSERT).